### PR TITLE
77125g update married applicant flow

### PR DESCRIPTION
**Pull request by michaelclement**
## Summary

This PR adds a new conditional screen that clarifies the circumstances surrounding an applicant's marriage to the sponsor if the sponsor is deceased.

- Added a new page
- Fixed a small bug where multiple relationship options could have the same 'spouse' option
- Updated conditional page flow

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/77125

## Testing done

- Manual testing
- Verified existing tests run and pass

## Screenshots

|         | Sponsor living | Sponsor deceased |
| ------- | ------ | ----- |
| First Screen |Screenshot 2024-03-14 at 12 29 28|Screenshot 2024-03-14 at 12 30 24|
| Second Screen | NA ||

## What areas of the site does it impact?

This form only

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [X] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [X] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [X] Browser console contains no warnings or errors.
- [X] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

NA

### CODE REVIEW

**Comment by michaelclement** [comment]
Waiting on https://github.com/department-of-veterans-affairs/vets-website/pull/28552 before opening this to review.

File: src/applications/ivc-champva/10-10D/config/form.js [hunk 1]
```diff
@@ -86,8 +86,17 @@ import ApplicantOhiStatusPage, {
...
-import { hasReq } from '../components/File/MissingFileOverview';
+import { MissingFileConsentPage } from '../pages/MissingFileConsentPage';
+import {
+  ApplicantRelOriginPage,
+  ApplicantRelOriginReviewPage,
+} from '../pages/ApplicantRelOriginPage';
+import {
+  ApplicantSponsorMarriageDetailsPage,
+  ApplicantSponsorMarriageDetailsReviewPage,
+} from '../pages/ApplicantSponsorMarriageDetailsPage';
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/config/form.js:97]: Reordering some imports and adding new custom page.

File: src/applications/ivc-champva/10-10D/config/form.js [hunk 2]
```diff
@@ -1122,6 +1126,39 @@ const formConfig = {
...
+                otherRelationshipToVeteran: { type: 'string' },
+              },
+            },
+          }),
+          uiSchema: {
+            applicants: {
+              items: {},
+            },
+          },
+        },
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/config/form.js:1161]: Add new custom page that only displays when sponsor is deceased and applicant was married to sponsor at some point. This is where we will collect additional details about the marriage to determine if they need to give us a certificate.

File: src/applications/ivc-champva/10-10D/config/form.js [hunk 3]
```diff
@@ -1133,7 +1170,18 @@ const formConfig = {
...
+                [
+                  'marriedTillDeathNoRemarriage',
+                  'marriedTillDeathRemarriedAfter55',
+                ].includes(
+                  get(
+                    'applicantSponsorMarriageDetails.relationshipToVeteran',
+                    formData?.applicants?.[index],
+                  ),
+                )) ||
+                !get('sponsorIsDeceased', formData))
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/config/form.js:1184]: We show this page IF:
- The applicant is currently married to the living sponsor
OR
- The applicant was married to the sponsor till the sponsor died.

File: src/applications/ivc-champva/10-10D/pages/ApplicantRelOriginPage.jsx [hunk 4]
```diff
@@ -66,7 +66,7 @@ export function ApplicantRelOriginPage(props) {
   };
   return ApplicantRelationshipPage(newProps);
 }
-export function ApplicationRelOriginReviewPage(props) {
+export function ApplicantRelOriginReviewPage(props) {
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/pages/ApplicantRelOriginPage.jsx:69]: Fixing a previous typo in the name of this component - also adjusted wherever it is imported, and internal to the component itself.

File: src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx [hunk 5]
```diff
@@ -65,7 +65,10 @@ function generateOptions({ data, pagePerItemIndex }) {
     marriageOptions.push({ label: marriedDeceased, value: 'spouse' });
   } else {
     marriageOptions.push({ label: marriedLiving, value: 'spouse' });
-    marriageOptions.push({ label: marriedLivingDivorced, value: 'spouse' });
+    marriageOptions.push({
+      label: marriedLivingDivorced,
+      value: 'spouseSeparated',
+    });
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx:71]: This is a bug fix. Previously, two objects each with value 'spouse' were pushed into the array, which caused rendering issues. This fixes that.

File: src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx [hunk 6]
```diff
@@ -253,9 +257,12 @@ export default function ApplicantRelationshipPage({
...
           class="vads-u-margin-y--2"
-          label={`What ${data.sponsorIsDeceased ? 'was' : 'is'} ${
-            useFirstPerson ? `your` : `${applicant}’s`
-          } relationship to the ${personTitle}?`}
+          label={
+            description ||
+            `What ${data.sponsorIsDeceased ? 'was' : 'is'} ${
+              useFirstPerson ? `your` : `${applicant}’s`
+            } relationship to the ${personTitle}?`
+          }
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/pages/ApplicantRelationshipPage.jsx:265]: Allows for a totally custom block of text to override the current subtitle text in the relationship component.

File: src/applications/ivc-champva/10-10D/pages/ApplicantSponsorMarriageDetailsPage.jsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import ApplicantRelationshipPage, {
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-10D/pages/ApplicantSponsorMarriageDetailsPage.jsx:1]: Essentially just a wrapper around `ApplicantRelationshipPage`, with a function to generate our custom-worded radio options that get fed into that component.

**Review by balexandr — APPROVED** [review]
Lgtm

**Review by reiting — APPROVED** [review]

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**